4c1d v / BE 003: Wallet Signature Verification

4c1d-V opened the pull request:
# Pull Request — Secure Nonce-Based Wallet Verification

---

## Summary
This PR introduces a new, secure wallet verification system for Starknet wallets. It implements:

- A nonce-based authentication
- Rate limiting
- JWT issuance
- Replay attack prevention

---

## Security Features
- Nonce is one-time use and expires after 5 minutes  
- Input format validation for all fields  
- Replay attack prevention via `invalidateNonce()`  
- Rate limiting added (10 req/15 min per IP)  

---

## Modified / Added
```bash
+ src/routes/v1/auth.ts             # Main secure verification endpoints
+ src/utilities/nonceManager.ts     # Manages nonce lifecycle
+ src/utilities/verifySignature.ts  # Validates wallet signatures
```
---

## Tests (Included)
- nonceManager.test.js
- verifySignature.test.js

---

jedstroke commented on backend/src/utilities/verifySignature.ts:
```diff
@@ -0,0 +1,39 @@
+import { verifyMessage } from "ethers";
```
Do `ethers` work with Starknet?

jedstroke replied:
ethers’ `verifyMessage` only works with the **ECDSA/secp256k1** signatures used on Ethereum‐compatible (EVM) chains. StarkNet uses a completely different curve (the Stark-friendly Pedersen hash + a different elliptic curve), so you can’t just drop ethers into a StarkNet app and expect signature verification to work.

Also, I don't remember building for signatures used on Ethereum‐compatible (EVM) chains. We are building on Starknet

jedstroke replied:
Aren't you supposed to use `Starknet.js` instead of `ethers`?

jedstroke commented on backend/tests/verifySignature.test.js:
I don't see a solid test case for Starknet?

jedstroke reviewed (changes requested):
@4c1d-V attend to these reviews

4c1d-V commented:
Hi, I've reviewed and updated accordingly, also added comprehensive test cases for both verifySignature and nonceManager. 

Also fixed the import path of types since the current one wouldn't reach ../src/types/ hope it's okay

Let me know if there's anything else

jedstroke commented:
@4c1d-V, Lemme see a Loom of you running the tests.

jedstroke commented on backend/tests/verifySignature.test.js:
```diff
@@ -0,0 +1,212 @@
+import { describe, it, expect, jest } from '@jest/globals';
+import { hash, ec, Signature } from 'starknet.js';
+import { verifyStarknetSignature } from '../verifySignature.starknet';
```
What are you referencing here?

jedstroke replied:
`'../verifySignature.starknet';` what is this?

jedstroke commented on backend/tests/verifySignature.test.js:
```diff
@@ -0,0 +1,212 @@
+      verify: jest.fn()
+    }
+  }
+}));
```
Why are you `starknet.js` library dependencies? Are you supposed to test with the dependencies?

jedstroke replied:
Now I am feeling like you GPTed everything headlessly.
